packages/backend/src/webhook/service.ts

```diff
@@ -132,44 +148,44 @@ async function processNextWebhookEvent(
 
   const tracer = trace.getTracer('webhook_worker')
 
-  return tracer.startActiveSpan(
-    'processNextWebhookEvent',
-    async (span: Span) => {
-      return deps_.knex!.transaction(async (trx) => {
-        const now = Date.now()
-        const events = await WebhookEvent.query(trx)
-          .limit(1)
-          // Ensure the webhook event cannot be processed concurrently by multiple workers.
-          .forUpdate()
-          // If a webhook event is locked, don't wait — just come back for it later.
-          .skipLocked()
-          .whereRaw(
-            `attempts < coalesce((select value from "tenantSettings" where "tenantId" = "webhookEvents"."tenantId" and key = '${TenantSettingKeys.WEBHOOK_MAX_RETRY.name}')::integer, ${deps_.config.webhookMaxRetry})`
-          )
-          .where('processAt', '<=', new Date(now).toISOString())
-
-        const event = events[0]
-        if (!event) return
-
-        const deps = {
-          ...deps_,
-          knex: trx,
-          logger: deps_.logger.child({
-            event: event.id
-          })
-        }
-
-        const settings = await deps_.tenantSettingService.get({
-          tenantId: event.tenantId
+  return tracer.startActiveSpan('processNextWebhook', async (span: Span) => {
+    return deps_.knex!.transaction(async (trx) => {
+      const now = Date.now()
+      const webhooks = await Webhook.query(trx)
+        .limit(1)
+        // Ensure the webhook cannot be processed concurrently by multiple workers.
+        .forUpdate()
+        // If a webhook is locked, don't wait — just come back for it later.
+        .skipLocked()
+        .whereRaw(
+          `attempts < GREATEST(coalesce((select value from "tenantSettings" where "tenantId" = "webhooks"."recipientTenantId" and key = '${TenantSettingKeys.WEBHOOK_MAX_RETRY.name}')::integer, ${deps_.config.webhookMaxRetry}))`
+        )
+        .where('processAt', '<=', new Date(now).toISOString())
+        .withGraphFetched('event')
+
+      const webhook = webhooks[0]
+      if (!webhook || !isWebhookWithEvent(webhook)) return
+
+      const deps = {
+        ...deps_,
+        knex: trx,
+        logger: deps_.logger.child({
+          event: webhook.eventId,
+          webhook: webhook.id
         })
-        const formattedSettings = formatSettings(settings)
+      }
 
-        await sendWebhookEvent(deps, event, formattedSettings)
-        span.end()
-        return event.id
+      const settings = await deps_.tenantSettingService.get({
+        tenantId: webhook.recipientTenantId
       })
-    }
-  )
+      const formattedSettings = formatSettings(settings)
+
+      await sendWebhookEvent(deps, webhook, formattedSettings)
```

Comment:
do we need to pass `operatorSettings` in? Otherwise i dont think we ever do the duplicate publish to operator?

Reply:
So `sendWebhookEvent` does not actually distinguish between if a webhook is for an operator or a tenant. Rather, webhooks for tenants and the operator are all occupying the same queue that `sendWebhookEvent` processes. So each time another service determines that a webhook event has occurred, it will create 1-2 webhooks depending on if the recipient is the operator or not.

This _does_ conflict with the concept I'm introducing in this PR which is the distinction between _webhooks_ and _webhook events_. This would probably make more sense if it were renamed to `processWebhook` instead of `processWebhookEvent`.

Reply:
as discussed in slack, it looks like it the `operatorSettings` just needs to be removed from `sendWebhookEvent` then